public/app/features/trails/AutomaticMetricQueries/previewPanel.ts

```diff
@@ -0,0 +1,48 @@
+import { SceneCSSGridItem, SceneQueryRunner, SceneVariableSet } from '@grafana/scenes';
+import { PromQuery } from 'app/plugins/datasource/prometheus/types';
+
+import { SelectMetricAction } from '../SelectMetricAction';
+import { hideEmptyPreviews } from '../hideEmptyPreviews';
+import { getVariablesWithMetricConstant, trailDS } from '../shared';
+import { getColorByIndex } from '../utils';
+
+import { getAutoQueriesForMetric } from './AutoQueryEngine';
+
+export function getPreviewPanelFor(metric: string, index: number, currentFilterCount: number) {
+  const autoQuery = getAutoQueriesForMetric(metric);
+
+  const vizPanel = autoQuery.preview
+    .vizBuilder()
+    .setColor({ mode: 'fixed', fixedColor: getColorByIndex(index) })
+    .setHeaderActions(new SelectMetricAction({ metric, title: 'Select' }))
+    .build();
+
+  const queries = autoQuery.preview.queries.map((query) =>
+    convertPreviewQueriesToIgnoreUsage(query, currentFilterCount)
+  );
+
+  return new SceneCSSGridItem({
```

Comment:
Replacing the `queries` with the result of this map is the key difference in this function.